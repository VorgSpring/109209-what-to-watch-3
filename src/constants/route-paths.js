export const RoutePaths = {
  MAIN: `/`,
  SING_IN: `/login`,
  FILM: `/film/:id`,
  REVIEW: `/film/:id/review`,
  FAVORITE: `/mylist`
};
