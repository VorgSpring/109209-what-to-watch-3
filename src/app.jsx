import React from 'react';
import {Provider} from 'react-redux';
import {BrowserRouter, Route, Switch} from 'react-router-dom';
import store from './store';
import {MainPage} from './components/main-page/main-page.jsx';
import {SingInPageContainer} from './components/sign-in-page/sign-in-page.jsx';
import {MoviePageContainer} from './components/movie-page/movie-page.jsx';
import {MoviePlayerContainer} from './components/movie-player/movie-player.jsx';
import {RoutePaths} from './constants/route-paths';

export const App = () => (
  <Provider store={store}>
    <BrowserRouter>
      <Switch>
        <Route
          exact
          path={RoutePaths.MAIN}
          component={MainPage}
        />

        <Route
          exact
          path={RoutePaths.FILM}
          component={MoviePageContainer}
        />

        <Route
          exact
          path={RoutePaths.SING_IN}
          component={SingInPageContainer}
        />
      </Switch>

      <MoviePlayerContainer />
    </BrowserRouter>
  </Provider>
);
