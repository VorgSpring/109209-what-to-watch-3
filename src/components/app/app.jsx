import React, {Fragment} from 'react';
import {Route, Switch} from 'react-router-dom';
import {MainPage} from '../main-page/main-page.jsx';
import {SingInPageContainer} from '../sign-in-page/sign-in-page.jsx';
import {MoviePageContainer} from '../movie-page/movie-page.jsx';
import {MoviePlayerContainer} from '../movie-player/movie-player.jsx';
import {RoutePaths} from '../../constants/route-paths';

export const App = () => (
  <Fragment>
    <Switch>
      <Route
        exact
        path={RoutePaths.MAIN}
        component={MainPage}
      />

      <Route
        exact
        path={RoutePaths.FILM}
        component={MoviePageContainer}
      />

      <Route
        exact
        path={RoutePaths.SING_IN}
        component={SingInPageContainer}
      />
    </Switch>

    <MoviePlayerContainer />
  </Fragment>
);
