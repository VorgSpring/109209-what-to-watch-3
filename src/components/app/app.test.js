import React from 'react';
import {MemoryRouter} from 'react-router-dom';
import {App} from './app.jsx';
import renderer from 'react-test-renderer';
import {RoutePaths} from '../../constants/route-paths.js';

jest.mock(`../main-page/main-page.jsx`, () => ({
  MainPage() {
    return <main-page />;
  }
}));

jest.mock(`../sign-in-page/sign-in-page.jsx`, () => ({
  SingInPageContainer() {
    return <sing-in-page />;
  }
}));

jest.mock(`../movie-page/movie-page.jsx`, () => ({
  MoviePageContainer() {
    return <movie-page />;
  }
}));

jest.mock(`../movie-player/movie-player.jsx`, () => ({
  MoviePlayerContainer() {
    return <movie-player />;
  }
}));

jest.mock(`../add-review-page/add-review-page.jsx`, () => ({
  AddReviewPageContainer() {
    return <add-review-page />;
  }
}));

jest.mock(`../favorite-movies-page/favorite-movies-page.jsx`, () => ({
  FavoriteMoviesPage() {
    return <favorite-movies-page />;
  }
}));

describe(`App`, () => {
  it(`should renders main-page`, () => {
    const tree = renderer.create(
        <MemoryRouter initialEntries={[RoutePaths.MAIN]}>
          <App />
        </MemoryRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`should renders movie-page`, () => {
    const tree = renderer.create(
        <MemoryRouter initialEntries={[RoutePaths.FILM]}>
          <App />
        </MemoryRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`should renders sing-in-page`, () => {
    const tree = renderer.create(
        <MemoryRouter initialEntries={[RoutePaths.SING_IN]}>
          <App />
        </MemoryRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`should renders add-review-page`, () => {
    const tree = renderer.create(
        <MemoryRouter initialEntries={[RoutePaths.REVIEW]}>
          <App />
        </MemoryRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`should renders favorite-movies-page`, () => {
    const tree = renderer.create(
        <MemoryRouter initialEntries={[RoutePaths.FAVORITE]}>
          <App />
        </MemoryRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
