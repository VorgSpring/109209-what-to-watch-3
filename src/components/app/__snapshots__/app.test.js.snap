// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should renders main-page 1`] = `
Array [
  <main-page />,
  <movie-player />,
]
`;

exports[`App should renders movie-page 1`] = `
Array [
  <movie-page />,
  <movie-player />,
]
`;

exports[`App should renders sing-in-page 1`] = `
Array [
  <sing-in-page />,
  <movie-player />,
]
`;
